ntbs-service/Pages/ContactDetails/Edit.cshtml.cs

```diff
@@ -65,6 +65,11 @@ public async Task<IActionResult> OnPostAsync()
                 return StatusCode((int)HttpStatusCode.Forbidden);
             }
 
+            if (UserId != ContactDetails.Id)
```

Comment:
This isn't quite what we want to achieve here - because a malicious user could still edit the request so that the `ContactDetails.Id` matches the url, but the username doesn't match.

I think instead we need to fetch the user from the user ID and then compare the username on the record retrieved from the database with the `ContactDetails.Username` provided on the request.

Reply:
Alright, I have made this change now

Reply:
This change caught a test which had been written incorrectly